Servers/Administration/OJS.Servers.Administration/Controllers/ContestsController.cs

```diff
@@ -111,32 +123,20 @@ public IActionResult Problems([FromQuery] IDictionary<string, string> complexId)
             EntityAction action,
             IDictionary<string, string> entityDict)
         {
-            if (entity.CategoryId.HasValue)
+            if (action == EntityAction.Create)
             {
-                await this.contestCategoriesValidationHelper
-                    .ValidatePermissionsOfCurrentUser(entity.CategoryId.Value)
-                    .VerifyResult();
+                return;
             }
```

Comment:
Why this check is removed? Lecturer should NOT be able to create contest in a category for which does not have permissions.

Reply:
It is changed because `ValidatePermissionsOfCurrentUser` method checks the user category permissions

Reply:
It is removed because it is incorrect. The user will see only allowed categories in the category dropdown in create/edit.